controllers/remote/cluster.go

```diff
@@ -38,11 +37,7 @@ func NewClusterClient(ctx context.Context, c client.Client, cluster *clusterv1.C
 	if err != nil {
 		return nil, err
 	}
-	mapper, err := apiutil.NewDynamicRESTMapper(restConfig, apiutil.WithLazyDiscovery)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create a DynamicRESTMapper for Cluster %s/%s", cluster.Namespace, cluster.Name)
-	}
-	ret, err := client.New(restConfig, client.Options{Scheme: scheme, Mapper: mapper})
```

Comment:
This is causing the test failure because the test is trying to verify a non-error result from `NewClusterClient`, which currently wouldn't happen due to the test not providing a server for the client to connect to.

Do we have a need for a long lived remote client that would necessitate lazy discovery? I'm not sure we do.